test/dummy/app/tasks/maintenance/params_task.rb

```diff
@@ -20,24 +20,36 @@ class ParamsTask < MaintenanceTasks::Task
 
     # Dropdown options with supported scenarios
     attribute :integer_dropdown_attr, :integer
+    attribute :integer_dropdown_attr_proc_no_arg, :integer
+    attribute :integer_dropdown_attr_from_method, :integer
     attribute :boolean_dropdown_attr, :boolean
 
     validates_inclusion_of :integer_dropdown_attr, in: [100, 200, 300], allow_nil: true
+    validates_inclusion_of :integer_dropdown_attr_proc_no_arg, in: proc { [100, 200, 300] }, allow_nil: true
     validates_inclusion_of :boolean_dropdown_attr, within: [true, false], allow_nil: true
+    validates_inclusion_of :integer_dropdown_attr_from_method, in: :dropdown_attr_options, allow_nil: true
 
     # Dropdown options with unsupported scenarios
-    attribute :text_integer_attr, :integer
-    attribute :text_integer_attr2, :integer
-    attribute :text_integer_attr3, :integer
+    attribute :text_integer_attr_proc_arg, :integer
+    attribute :text_integer_attr_unbounded_range, :integer
+    attribute :text_integer_attr_method_arg, :integer
 
-    validates_inclusion_of :text_integer_attr, in: proc { [100, 200, 300] }, allow_nil: true
-    validates_inclusion_of :text_integer_attr2, in: :undefined_symbol, allow_nil: true
-    validates_inclusion_of :text_integer_attr3, in: (100..), allow_nil: true
+    validates_inclusion_of :text_integer_attr_proc_arg, in: proc { |_task| [100, 200, 300] }, allow_nil: true
+    validates_inclusion_of :text_integer_attr_unbounded_range, in: (100..), allow_nil: true
+    validates_inclusion_of :text_integer_attr_method_arg, in: :method_with_arg, allow_nil: true
```

Comment:
This will raise `ArgumentError` if a value is supplied in the UI and the task is ran, because the inclusion validator for Active Model Attributes does not support methods with arguments.

I'd still like to test that the UI renders properly -- that the arity check works and we render a text field here instead of a dropdown.

Reply:
We shouldn't keep it because it's not a valid Active Record validation.